reconcile/openshift_namespaces.py

```diff
@@ -39,29 +39,29 @@ def get_desired_state(
     return desired_state
 
 
-def get_shard_namespaces(namespaces: Any) \
-        -> Tuple[List[Dict[str, str]], bool]:
+def get_shard_namespaces(
+        namespaces: Iterable[Mapping[str, Any]]) \
+            -> Tuple[List[Dict[str, str]], bool]:
 
     # Structure holding duplicates by namespace key
     duplicates: Dict[str, List[Dict[str, str]]] = {}
     # namespace filtered list without duplicates
-    filtered_ns: Dict[str, Dict[str, str]] = {}
+    filtered_ns: Dict[str, Dict[str, Any]] = {}
 
     err = False
     for ns in namespaces:
         key = f'{ns["cluster"]["name"]}/{ns["name"]}'
 
         if is_in_shard(key):
             if key not in filtered_ns:
-                filtered_ns[key] = ns
+                filtered_ns[key] = dict(ns)
```

Comment:
Why this extra copy? If the type checker was giving you headaches, `cast` would be a better option wit no run-time cost:

```python
from typing import cast
...
filtered_ns[key] = cast(dict, ns)
```

Reply:
👍